profiles/source_profile.go

```diff
@@ -409,13 +410,91 @@ func NewSourceProfileCsv(params map[string]string) SourceProfileCsv {
 	return csvProfile
 }
 
+type SourceProfileStreamingMySQL struct {
+	Conn            SourceProfileConnectionMySQL
+	StreamingConfig string
+}
+
+func NewSourceProfileStreamingMySQL(params map[string]string) (SourceProfileStreamingMySQL, error) {
+	profile := SourceProfileStreamingMySQL{}
+	if profile.StreamingConfig = params["streamingCfg"]; profile.StreamingConfig == "" {
+		return profile, fmt.Errorf("specify a non-empty streaming config file path")
+	}
+	conn := SourceProfileConnectionMySQL{}
+
+	host, hostOk := params["host"]
```

Comment:
We could but that will also allow for environment variables for mysql in subcommand mode. I don't think we should allow that to keep it consistent. We can refactor certain aspects though because certain if else logic is reused across all engines.